engine/gui/src/gui.cpp

```diff
@@ -740,6 +742,28 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
         return *font;
     }
 
+    Result AddMaterial(HScene scene, const char* material_name, void* material)
+    {
+        if (scene->m_Materials.Full())
+        {
+            return RESULT_OUT_OF_RESOURCES;
+        }
+
+        uint64_t name_hash = dmHashString64(material_name);
+        scene->m_Materials.Put(name_hash, (dmRender::HMaterial) material);
+
+        InternalNode* nodes = scene->m_Nodes.Begin();
+        for (uint32_t i = 0; i < scene->m_Nodes.Size(); ++i)
+        {
+            if (nodes[i].m_Node.m_MaterialNameHash == name_hash)
+            {
+                nodes[i].m_Node.m_Material = material;
+            }
+        }
```

Comment:
I think rather than looping for every node, for each material, we could have a loop outside of the function (at setup) where we lookup the materials from the map instead?